src/clean_air/models.py

```diff
@@ -27,21 +34,879 @@ class ContactDetails:
+
+        # We can't convert from months to days (or other things) except for converting 12 months to a year.
+        # So we can use this fact to convert as many months to years as possible, then we can convert years to seconds
+        # which gives us greater flexibility when the output of this
+        if months:
+            # We can convert multiples of 12 months to years because we can guarantee that 12 months always equals 1
+            # year, but can't convert partial months because different months have different lengths, so can't guarantee
+            # their length
+            years_from_months = int(months / Duration._MONTHS_IN_YEAR)
+            if years_from_months:
+                years = years + years_from_months
+                # Note, math.fmod is not the same as the modulo operator,
+                # refer to https://docs.python.org/3/library/math.html#math.fmod
+                months = int(math.fmod(months, Duration._MONTHS_IN_YEAR))  # preserves the sign
+
+        seconds += (
+                minutes * Duration._SECONDS_IN_MINUTE + hours * Duration._SECONDS_IN_HOUR
+                + days * Duration._SECONDS_IN_DAY + weeks * Duration._SECONDS_IN_WEEK
+                + years * Duration._SECONDS_IN_YEAR
+        )
+
+        return months, seconds
+
+    @staticmethod
+    def _to_dict(obj: Union["Duration", relativedelta, timedelta]) -> Dict[str, float]:
+        """
+        Helper method used internally, particularly for situations where we're still calculating values for
+        creating a new Duration instance. Specifically useful for situations where fields contain negative values that
+        we're resolving
+        """
+        if isinstance(obj, Duration):
+            return {
+                "years": obj.years,
+                "months": obj.months,
+                "days": obj.days,
+                "hours": obj.hours,
+                "minutes": obj.minutes,
+                "seconds": obj.seconds,
+                "weeks": obj.weeks
+            }
+        elif isinstance(obj, relativedelta):
+            kwargs = {
+                "years": obj.years,
+                "months": obj.months,
+                "days": obj.days,
+                "hours": obj.hours,
+                "minutes": obj.minutes,
+                "seconds": obj.seconds + obj.microseconds / Duration._MICROSECONDS_IN_SECOND
+            }
+
+            # relativedelta doesn't distinguish between 0 and unset, so we'll treat 0 as unset, and filter out anything
+            # without a value
+            kwargs = {k: v for k, v in kwargs.items() if v}
+            # relative deltas can contain negative fields, so we need to normalise the result
+            return Duration._normalise(**kwargs)
+        elif isinstance(obj, timedelta):
+            return Duration._normalise(seconds=obj.total_seconds())
+        else:
+            raise NotImplementedError(f"{type(obj)} is not supported")
+
+    @staticmethod
```

Comment:
Essentially the rationale for making them static is that they don't access any instance attributes.
The reason for that is that they're used in methods like `__add__` and `__sub__` when we're doing arithmetic with things like `datetime`, `timedelta`, and `relativedelta`. I.e. we don't actually have a `Duration` object but still need to perform the calculation. It also means I don't have to create multiple `Duration` objects, and in particular there are cases where I can't create a Duration until after I've normalised the values.

e.g. relativedelta supports negatives where Duration doesn't, which causes problems in cases like this:
`Duration(years=1) + relativedelta(days=-1)`. If we don't normalise the fields, we end up trying to create `Duration(years=1, days=-1)` which throws a `ValueError`. Normalisation converts that to `Duration(days=364)`.

Reply:
I guess what I was thinking was that if `__new__` was used to encapsulate the pre-processing necessary then we'd always have valid Duration instances and fewer intermediates - and, in my mind at least, it would feel more Pythonic. But that's an implementation detail, and `@staticmethod` is pretty Pythonic!

Reply:
Yea, I did have that thought, but I wanted to avoid forcing all `Duration`s to be normalised, as then the parsing function would become lossy/would no longer represent the duration string as given. I can imagine cases where someone might want to keep `P36H` as `Duration(hours=36)` rather than be normalised to `Duration(days=1, hours=12)`. E.g. when dealing with timesteps relating to model runs they tend to be all expressed in un-normalised hours.

So it's not that I didn't want to do what you suggest, but it seemed incompatible with some of the (admittedly self-imposed) constraints on the behaviour!